### DISCUSSION TITLE
unable to read variable using env/strenv in a select

### DISCUSSION THREAD

**Discussion by traghu84** [mikefarah/yq#1573] [Q&A] [answered]
I have my yaml as below.

name: "MicroService"
_server:
  type: 'Apache'
  names:
    - 'ROW1'
    - 'ROW2'
    - 'ROW3'
    - 'NAFTA1'
    - 'NAFTA2'
    - 'NAFTA3'_

When i run this command below directly, i get the proper output.
 yq '.server.names[] | select( . != "*ROW*" )' configs.yaml
**output**:
NAFTA1
NAFTA2
NAFTA3

But if do the same by storing the select condition into a variable, its not working as expected.
REGION='. != "*ROW*"'
yq  '.server.names[] | select(strenv(REGION))' configs.yaml
**output**:
ROW1
ROW2
ROW3
NAFTA1
NAFTA2
NAFTA3

I tried using **env** as well, but the result is same.
By the way i am running this from a bash in jenkins pipeline.
Am i doing something wrong here? Can someone please help, this is the final step for completing my work.

**Comment by mikefarah** [comment, marked as answer]
`strenv` will read the variable as a string - looks like you want to evaluate that string as an expression - try `eval(strenv(REGION))` see https://mikefarah.gitbook.io/yq/operators/eval#dynamically-update-a-path-from-an-environment-variable for an example

**Reply by traghu84** [reply]
Thank you for the help @mikefarah , it solved my problem
